src/gettext.h

```diff
@@ -59,3 +59,21 @@ inline std::string strgettext(const std::string &text)
 {
 	return text.empty() ? "" : gettext(text.c_str());
 }
+
+/**
+ * Returns translated string with format args applied
+ *
+ * @tparam Args Template parameter for format args
+ * @param src Translation source string
+ * @param args Variable format args
+ * @return translated string
+ */
+template <typename ...Args>
+inline std::wstring fwgettext(const char *src, Args&&... args)
+{
+	wchar_t buf[255];
+	const wchar_t* str = wgettext(src);
+	swprintf(buf, sizeof(buf) / sizeof(wchar_t), str, std::forward<Args>(args)...);
+	delete[] str;
+	return std::wstring(buf);
```

Comment:
You could fix the hardcoded limit like this:
```c++
    std::wstring ret;
    const wchar_t* str = wgettext(src);
    const int required_size = swprintf(nullptr, 0, str, std::forward<Args>(args)...);
    if (required_size > 0) {
        ret.resize(required_size);
        swprintf(&ret[0], ret.size(), str, std::forward<Args>(args)...);
    }
    delete[] str;
    return ret;
```

Reply:
According to https://en.cppreference.com/w/cpp/io/c/fwprintf:
> ### Notes
>While narrow strings provide std::snprintf, which makes it possible to determine the required output buffer size, there is no equivalent for wide strings, and in order to determine the buffer size, the program may need to call std::swprintf, check the result value, and reallocate a larger buffer, trying again until successful.

Also, I think you'd have an off-by-one error, as you need a size of 1 greater that `reqired_size` (for the 0).

(Oh, and you could wrap `str` into a `std::unique_ptr<wchar_t[]>`.)

Reply:
Hmm, I thought it would behave identically. That's unfortunate..